Final game.py

```diff
@@ -0,0 +1,349 @@
+
+    try:
+        choice = int(input("\nSelect a save file to load (Enter number): "))
+        if 1 <= choice <= len(save_files):
+            filename = save_files[choice - 1]
+            with open(filename, "r") as file:
+                data = json.load(file)
+                player = Player(
+                    name=data["name"], 
+                    inventory=data.get("inventory", []),
+                    health=data.get("health", 100),
+                    level=data.get("level", 1),
+                    credits=data.get("credits", 100)
+                )
+            print("\nGame loaded sucussefully from {filename}.")
+            return player
+        else:
+            print("\nInvalid selection. Returning to main menu.")
+            return None
+    except ValueError:
+        print("\nNo saved game found. Returning to main menu")
+        return None
+    except Exception as e:
+        print(f"\nError loading game: {e}")
+        return None
+#to load in previously saved inventory        
+def load_inventory():
+    try:
+        with open('player_inventory.json', 'r') as file:
+            inventory = json.load(file)
+        return inventory
+    except FileNotFoundError:
+        print("Inventory file not found! Creating a new inventory.")
+        return {"items": []}
+
+# Function to save inventory data to JSON file
+def save_inventory(inventory):
+    with open('player_inventory.json', 'w') as file:
+        json.dump(inventory, file)
+    print("Inventory saved.")
+
+#this where it shows player status and game menu options
+def play_game(player, galaxy, shop):
+    while True:
+        print("\n=== Player Status ===")
+        print(f"Name: {player.name}")
+        print(f"Health: {player.health}")
+        print(f"level: {player.level}")
+        print(f"Credits: {player.credits}")
+        print("\n--- Game Menu ---")
+        print("1. Show Inventory")
+        print("2. Explore Galaxy")
+        print("3. Save Game")
+        print("4. Save Inventory")
+        print("5. Exit to Main Menu")
+        print("6. Visit Space Shop")
+
+        choice = input("Choose an option: ")
+        if choice =="1":
+            player.show_inventory()
+        elif choice == "2":
+            galaxy.show_planets()
+            try:
+                planet_choice = int(input("Select a planet to explore:  "))
+                if 1 <= planet_choice <= len(galaxy.planets):
+                    galaxy.planets[planet_choice - 1].explore(player)
+                else:
+                        print("\nInvalid planet selection.")
+            except ValueError:
+                    print("\nPlease enter a valid number.")
+        elif choice == "3":
+            save_character(player)
+        elif choice == "4":
+            save_inventory(player.to_dict())
+        elif choice == "5":
+            print("\nReturning to the main menu..")
+            break
+        elif choice == "6": 
+            shop.buy_item(player)
```

Comment:
Maybe add the shop option before the save and exit option.

```suggestion
        if choice =="1":
            player.show_inventory()
        elif choice == "2":
            galaxy.show_planets()
            try:
                planet_choice = int(input("Select a planet to explore:  "))
                if 1 <= planet_choice <= len(galaxy.planets):
                    galaxy.planets[planet_choice - 1].explore(player)
                else:
                        print("\nInvalid planet selection.")
            except ValueError:
                    print("\nPlease enter a valid number.")
        elif choice == "3": 
            shop.buy_item(player)
        elif choice == "4":
            save_character(player)
        elif choice == "5":
            save_inventory(player.to_dict())
        elif choice == "6":
            print("\nReturning to the main menu..")
            break
        
```